third_party/ngen/ngen_asm.hpp

```diff
@@ -1,6 +1,6 @@
 #ifndef NGEN_ASM_HPP
 #define NGEN_ASM_HPP
-#if NGEN_ASM
+#ifdef NGEN_ASM
 
```

Comment:
Coincidentally, this change also appeared #2773, so there is not need to port this to main.

Reply:
@rjoursler  Could you cherry-pick #2773 to rls-v3.8-pc branch? I'm integrating rls-v3.8-pc branch to public OpenVINO.

Reply:
I don't think backporting #2773 makes sense in this case. That change is relatively invasive, so keeping the simple fix provided here is much better. I can go ahead and take over getting this merged in #2824 so that we can get this merged soon.